TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/HttpJsonMessageWithFaultingPayload.java

```diff
@@ -48,11 +49,24 @@ public void setProtocol(String value) {
     }
 
     @Override
-    public Map<String, Object> headersMap() {
-        return Collections.unmodifiableMap(headers());
+    public Map<String, List<String>> headers() {
+        Map<String, Object> originalHeaders = this.headersInternal();
+        Map<String, List<String>> convertedHeaders = new LinkedHashMap<>();
+
+        for (Map.Entry<String, Object> entry : originalHeaders.entrySet()) {
+            if (entry.getValue() instanceof List) {
+                @SuppressWarnings("unchecked")
+                List<String> values = (List<String>) entry.getValue();
+                convertedHeaders.put(entry.getKey(), values);
+            } else if (entry.getValue() != null) {
+                convertedHeaders.put(entry.getKey(), Collections.singletonList(entry.getValue().toString()));
+            }
+        }
+
+        return Collections.unmodifiableMap(convertedHeaders);
     }
 
-    public ListKeyAdaptingCaseInsensitiveHeadersMap headers() {
+    public ListKeyAdaptingCaseInsensitiveHeadersMap headersInternal() {
```

Comment:
Following through from the last comment - I'm not sure why everybody in the replayer dealing with this object deals with headers through an 'internal' interface.  This should be the public one.  For adapting to other package boundaries, do that separate from the class.

Reply:
addressed